core/src/main/scala/org/apache/spark/storage/MemoryStore.scala

```diff
@@ -409,40 +397,42 @@ private[spark] class MemoryStore(blockManager: BlockManager, maxMemory: Long)
    * from the same RDD (which leads to a wasteful cyclic replacement pattern for RDDs that
    * don't fit into memory that we want to avoid).
    *
-   * Assume that `accountingLock` is held by the caller to ensure only one thread is dropping
```

Comment:
yes, `ensureFreeSpace` now explicitly holds the `accountingLock`, since it can be called from random places like our memory manager